sdk/src/main/java/io/opentelemetry/sdk/common/export/CompletableResultCode.java

```diff
@@ -0,0 +1,70 @@
+package io.opentelemetry.sdk.common.export;
+
+import java.util.concurrent.Executor;
+
+/**
+ * The implementation of Export operations are often asynchronous in nature, hence the need to
+ * convey a result at a later time. CompletableResultCode facilitates this.
+ *
+ * <p>This class models JDK 8's CompletableFuture to afford migration should Open Telemetry's SDK
+ * select JDK 8 or greater as a baseline, and also to offer familiarity to developers.
+ */
+public class CompletableResultCode {
```

Comment:
We may as well make this generic

Reply:
This will add a lot of complexity, which I don’t think is warranted. Ultimately migrating to a CompleteableFuture approach for all async use cases would be best.

I’m not attempting to solve all Open Telemetry async use cases, and I want to ensure that the approach taken here remains reasonable in scope.